taichi/platform/metal/shaders/helpers.metal.h

```diff
@@ -33,6 +33,18 @@ STR(template <typename T, typename G> T union_cast(G g) {
       return (((lhs * rhs < 0) && (rhs * intm != lhs)) ? (intm - 1) : intm);
     }
 
+    int32_t pow_i32(int32_t x, int32_t n) {
+      int32_t tmp = x;
+      int32_t ans = 1;
+      while (n) {
+        if (n & 1)
+          ans *= tmp;
+        tmp *= tmp;
+        n >>= 1;
+      }
+      return ans;
+    }
```

Comment:
Maybe we also need a pow_i64?

Reply:
Sorry I didn't realize that Metal doesn't support i64...

Reply:
Yeah, really unfortunate :-/